src/Components/PostComment.jsx

```diff
@@ -0,0 +1,60 @@
+import { useState } from "react";
+import { postComment } from "./Api";
+
+export default function PostComment({ articleId }) {
+  const [username, setUsername] = useState("");
+  const [body, setBody] = useState("");
+  const [submitting, setSubmitting] = useState(false);
+  const [error, setError] = useState(null);
+  const [success, setSuccess] = useState(false);
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    if (!username || !body) {
+      setError("Please enter both username and comment.");
+      return;
+    }
+    setSubmitting(true);
+    setUsername("");
+    setBody("");
+    setError(null);
+    postComment(articleId, { username, body })
+      .then(() => {
+        setSuccess(true);
+      })
+      .catch((error) => {
+        setSuccess(false);
+        setError(
+          "An error occurred while posting your comment. Please try again later."
+        );
+        setSubmitting(false);
+      });
+  };
+  return (
+    <div>
+      <form onSubmit={handleSubmit} className="comment-form-container">
+        <input
+          type="text"
+          value={username}
+          onChange={(e) => setUsername(e.target.value)}
```

Comment:
Asking the user to manually input the _username_ poses an issue. They might not know what the valid usernames are, so an employer looking at this might not be able to leave a comment. My advise here is to have a default username for now which is provided, then you can revisit this when implementing a login / users page